### ISSUE TITLE
Unable to get headers and statuscode using the examples

### ISSUE THREAD

**Issue by Zeychiel** [krakend/krakend-ce#837]
**Environment info:**

* KrakenD version: 2.5.0
* System info: docker
* Backend technology: Go

**Describe what are you trying to do**:
I try to catch 201 responses from the backend to add a dedicated header ; using a response modifier
but
`.Headers()` and `.StatusCode()` on` input.(ResponseWrapper)` are nil ; `.Data()` on the other hand does as expected

This issue: https://github.com/krakend/krakend-ce/issues/488#issuecomment-1157921730 says that this is by design ; but this example from the documentation: https://www.krakend.io/docs/extending/plugin-modifiers/#building-the-plugin
![image](https://github.com/krakend/krakend-ce/assets/10617848/6fc64325-419c-4d3b-bfe6-852cf157212b)
says otherwise.

Is the documentation outdated and it is by design that the values are empty when NOT using no-op ?

Thanks,

**Comment by nsitbon** [comment]
Hi @Zeychiel you're right that's by design and yes the documentation is wrong.
Best regards.

**Comment by thedae** [comment]
Yes, the headers and status code are only available for `no-op` encoding, as pointed out in [the documentation](https://www.krakend.io/docs/extending/plugin-modifiers/#building-the-plugin) (it's been a while since this issue was opened, so my apologies if this warning was not present at the time):

> If your endpoint uses an output_encoding other than no-op you can work with ResponseWrapper.Data() and ResponseWrapper.IsComplete(). If you use no-op you can work with ResponseWrapper.Io(), ResponseWrapper.StatusCode(), and ResponseWrapper.Headers().
If you set data in a different place than specified above, it will be ignored.

The code example is just that, an example showcasing all the available methods.
